File: Frontend - Pedro Assumpção/Sistema_Funcionarios/src/app/models/funcionario.ts
export interface Funcionario {
  id: number;
  nome: string;
  email: string;
  cargo: string;
  salario: number;
  dataAdmissao: string;   // yyyy-MM-dd
  ativo: boolean;
}

export interface FuncionarioRequest {
  nome: string;
  email: string;
  cargo: string;
  salario: number;
  dataAdmissao: string;   // yyyy-MM-dd
}

export interface FuncionarioResponse extends Funcionario {}

export interface FuncionarioFiltro {
  cargo?: string;
  ativo?: boolean;
}

File: GestãoDepartamentoWagner/funcionarios-back-master/funcionarios-back-master/src/main/java/com/senai/jonatas/funcionarios/entity/Funcionario.java
package com.senai.jonatas.funcionarios.entity;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Table(name = "funcionarios", indexes = {@Index(name = "ix_funcionario_email", columnList = "email")})
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Funcionario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String nome;

    @Column(nullable = false, unique = true)
    private String email;

    @Column(nullable = false)
    private String cargo;

    @Column(nullable = false, scale = 2)
    private BigDecimal salario;

    @Column(nullable = false)
    private LocalDate dataAdmissao;

    @Column(nullable = false)
    private Boolean ativo = true;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "departamento_id")
    private Departamento departamento;
}
